### DISCUSSION TITLE
How do I install my application in the virtualenv?

### DISCUSSION THREAD

**Discussion by dsedivec** [pdm-project/pdm#2652] [Q&A] [answered]
Hi!  I'm trying to use PDM 2.12.3.  I have a *very* simple `pyproject.yaml`:

```
[project]
name = "myapp"
version = "0.1.0"
description = "Default template for PDM package"
authors = [
    {name = "me", email = "me@example.com"},
]
dependencies = [
]
requires-python = "==3.12.*"
readme = "README.md"
license = {text = "GPL-3.0-or-later"}

[project.scripts]
myapp = "myapp.__main__:main"

[tool.pdm]
distribution = false
```

I am developing from a virtualenv (created and activated by direnv, if that matters).  When I run e.g. `pdm install`, my application is not installed into the virtualenv.  If I `pip install -e .`, `myapp` will inevitably be removed by PDM, such as when I next run `pdm remove some_unrelated_dependency`.

What is the appropriate way to have PDM install my application into the virtualenv, and keep it there, so I can run `myapp` from the virtualenv during testing?

Thanks for writing PDM!

**Comment by frostming** [comment, marked as answer]
```toml
[tool.pdm]
distribution = true
```

It was determined by the `pdm init` which I think clearly says what it does:

```
Do you want to build this project for distribution(such as wheel)?
If yes, it will be installed by default when running `pdm install`. [y/n] (n): 
````

**Reply by dsedivec** [reply]
You're right.  I could have sworn it asked me something about "are you building a library", but of course you're right.  Sorry for the dumb question, thank you for the answer.
